### PR TITLE
Feature/get user

### PR DESCRIPTION
**Pull request by kyamatek**
Description: 

- Added a getting user data (id, username, email, hashed password) function by email

To test this pull request:

- Use postman to send a request to http://localhost:3000/user/getByEmail using raw JSON
- Fields to include when sending the request:
```
{
    "email":"xyz@zyx",
}
```

- xyz@zyx is the existing data in database
- e.g. mail

Success condition:
- Get user data by xyz@zyx

Sample output when succesfully getting user data:
```
{
    "success": true,
    "data": [
        {
            "id": "fb775a01-dba7-4dcc-ba47-251043ed9187",
            "username": "name",
            "email": "mail",
            "password": "$2a$10$xzn/W9604JrdcDNLKSWcqeF4aPoOovrLWBYdHBPkOF9J0K50N04TO"
        }
    ]
}
```

### CODE REVIEW

**Review by ShimizuKazuki — COMMENTED** [review]
I think there is nothing to fix.

File: back/src/database/UserSQL.js [hunk 1]
```diff
@@ -34,4 +34,25 @@ async function saveUser(newUser) {
     });
 }
 
-module.exports = {saveUser:saveUser};
\ No newline at end of file
+async function roadUserByEmail(email) {
```
Comment (ONourry) [inline back/src/database/UserSQL.js]: Let's change that function name to getUserByEmail(email)

File: back/src/database/UserSQL.js [hunk 2]
```diff
@@ -34,4 +34,25 @@ async function saveUser(newUser) {
...
+            return {"success":false, "data":exception};      
+        });
+    }).catch((exception)=>{
+        connection.end();
+        error(fileLabel,"Error trying to connect to database: " + exception);
+        return {"success":false, "data":exception};
+    });
+}
+
+module.exports = {saveUser:saveUser, roadUserByEmail:roadUserByEmail};
```
Comment (ONourry) [inline back/src/database/UserSQL.js]: Change to getUserByEmail here too

File: back/src/route/userController.js [hunk 3]
```diff
@@ -31,4 +31,21 @@ module.exports = class UserController{
             return res.status(400).json({"success": false, "error": exception});
         });
     }
+
+    async getByEmail(req, res){
```
Comment (ONourry) [inline back/src/route/userController.js]: getUserByEmail instead of getByEmail

File: back/src/route/userRoute.js [hunk 4]
```diff
@@ -4,5 +4,6 @@ const UserController = require('./userController');
 const userController = new UserController();
 
 router.post("/register",userController.register);
+router.get("/getByEmail",userController.getByEmail);
```
Comment (ONourry) [inline back/src/route/userRoute.js]: change route name to getUserByEmail

**Review by ONourry — CHANGES_REQUESTED** [review]

**Comment by ONourry** [comment]
The code currently returns the right user if it exists in the database but it does not handle a request where a user does not exist. If we query a user that does not exist, the return code is still success: true with empty data. The return code should be false with an error message saying that the user does not exist.

Screen Shot 2020-11-02 at 4 55 26 PM

**Comment by tsukasa2** [comment]
Test is succeeded

**Review by ONourry — APPROVED** [review]

**Pull request merged**